SwiftWisdom/Core/StandardLibrary/Array/Array+Utilities.swift

```diff
@@ -32,6 +32,37 @@ extension Array {
     }
 }
 
+extension Array {
+    /// Separates this array's elements into chunks with a given length.
+    ///
+    /// If `length` is not a positive number or if it's greater than or equal to this array's length, we return a new array
+    /// whose sole element is this array itself.
+    ///
+    /// - parameter length: Maximum number of elements in each chunk.
+    /// - parameter includeRemainder: If set to `true`, include the last chunk in returned array even if its length is less than
+    ///   `length`. If set to `false`, discard the last chunk if its length is less than `length`. Defaults to `true`.
+    ///
+    /// - returns: New array of chunks of elements in this array.
+    public func ip_chunks(ofLength length: Int, includeRemainder: Bool = true) -> [[Element]] {
+        guard length > 0 else { return [self] }
+
+        var mutable = self
+        var chunks = [[Element]]()
+
+        while mutable.count >= length {
+            let next = Array(mutable.prefix(length))
+            chunks.append(next)
+            mutable.removeFirst(length)
```

Comment:
To address your question about the `length` > `array.count` case - I think that the correct way to think of it is to just "do the math."  

In the "normal" case, you're returning `array.count / length` chunks, and then one additional chunk of the "leftovers" if (`array.count % length > 0` AND `includeRemainder == true`).

I think that's exactly what you should do in the `length` > `array.count` case.  In that case, `array.count / length` is zero, so you return zero chunks.  Then, you also return one more chunk of the "leftovers" (in this case, the entire array!) if (`array.count % length > 0` (which you know to be true!) AND `includeRemainder == true` )